chart/templates/etcd-bootstrap-configmap.yaml

```diff
@@ -9,23 +9,65 @@ metadata:
 data:
   bootstrap.sh: |-
     #!/bin/sh
-    while true;
-    do
-      wget http://localhost:8080/initialization/status -S -O status;
-      STATUS=`cat status`;
-      case $STATUS in
-      "New")
-            wget http://localhost:8080/initialization/start -S -O - ;;
-      "Progress")
-            sleep 1;
-            continue;;
-      "Failed")
-            continue;;
-      "Successful")
-            exec etcd --config-file /var/etcd/config/etcd.conf.yaml
-            ;;
-      esac;
-    done
+    VALIDATION_MARKER=/var/etcd/data/validation_marker
+    
+    trap_and_propagate() {
+        PID=$1
+        shift
+        for sig in "$@" ; do
+            trap "kill -$sig $PID" "$sig"
+        done
+    }
+    
+    start_managed_etcd(){
+          etcd --config-file /bootstrap/etcd.conf.yml &
+          ETCDPID=$!
+          trap_and_propagate $ETCDPID INT TERM
+          wait $ETCDPID
+          RET=$?
+          echo $RET > $VALIDATION_MARKER
+          exit $RET
+    }
+    
+    check_and_start_etcd(){
+          while true;
+          do
+            wget http://localhost:8080/initialization/status -S -O status;
+            STATUS=`cat status`;
+            case $STATUS in
+            "New")
+                  wget http://localhost:8080/initialization/start?mode=$1 -S -O - ;;
+            "Progress")
+                  sleep 1;
+                  continue;;
+            "Failed")
+                  continue;;
+            "Successful")
+                  start_managed_etcd
+                  break
+                  ;;
+            esac;
+          done
+    }
+    
+    if [ ! -f $VALIDATION_MARKER ] ;
+    then 
+          echo "No VALIDATION_MARKER file. Perform complete initialization routine and start and start etcd."
```

Comment:
1. replace VALIDATION_MARKER with $VALIDATION_MARKER. 
2. typo:  repeated "and start".